From bb9b2ed88c71414005f86b44ca94c0495de59722 Mon Sep 17 00:00:00 2001
From: khanhluu <khanh.luu.xw@rvc.renesas.com>
Date: Tue, 13 Feb 2018 02:10:32 +0700
Subject: [PATCH 300/629] drm/i2c: sii902x: Avoid appearing error EBUSY

Once the host has finished reading EDID, it must write TPI 0x1A = 00 to clear the request.

Signed-off-by: khanhluu <khanh.luu.xw@rvc.renesas.com>
Signed-off-by: vietn <vietn@fsoft.com.vn>
---
 drivers/gpu/drm/i2c/sii902x.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/gpu/drm/i2c/sii902x.c b/drivers/gpu/drm/i2c/sii902x.c
index 86f6a733b38e..29f4c1938885 100644
--- a/drivers/gpu/drm/i2c/sii902x.c
+++ b/drivers/gpu/drm/i2c/sii902x.c
@@ -122,7 +122,8 @@ static int sii902x_get_modes(struct drm_encoder *encoder,
 	cnt = 100;
 	do {
 		cnt--;
-		sii902x_write(client, 0x1A, old & ~0x6);
+		/*Once the host has finished reading EDID, it must write TPI 0x1A= 00 to clear the request*/
+		sii902x_write(client, 0x1A, 0x0);
 		msleep(10);
 		dat = sii902x_read(client, 0x1A);
 	} while ((dat & 0x6) && cnt);
-- 
2.17.1

